core/src/main/java/io/grpc/internal/SerializingExecutor.java

```diff
@@ -82,90 +69,48 @@ public SerializingExecutor(Executor executor) {
     this.executor = executor;
   }
 
-  private final Object internalLock = new Object() {
-    @Override public String toString() {
-      return "SerializingExecutor lock: " + super.toString();
-    }
-  };
-
   /**
    * Runs the given runnable strictly after all Runnables that were submitted
    * before it, and using the {@code executor} passed to the constructor.     .
    */
   @Override
   public void execute(Runnable r) {
-    Preconditions.checkNotNull(r, "'r' must not be null.");
-    boolean scheduleTaskRunner = false;
-    synchronized (internalLock) {
-      waitQueue.add(r);
+    runQueue.offer(checkNotNull(r, "'r' must not be null."));
```

Comment:
I can, though you might consider an alternative:
```
boolean success = runQueue.offer(r);
assert success;
schedule();
```

I haven't done testing but this will lightly inflate the method making it harder to inline.  `add()` is just a wrapper around `offer()`.  The reason I picked offer is because I am using the [analogous](https://docs.oracle.com/javase/7/docs/api/java/util/Queue.html) function `poll()` below.  


Thoughts?